### DISCUSSION TITLE
🚀[FEATURE]: Hook to specific lifecycle event in @Action()

### DISCUSSION THREAD

**Discussion by joaqcid** [ngxs/store#1930] [Ideas]
### Relevant Package
This feature request is for @ngxs/store

### Description
Be able to hook onto a specific lifecycle event of an Action 

### Describe the problem you are trying to solve
Right now, if you need to do something only when action is succesful or error or dispatched, you need to inject `Actions` stream and filter lifecyclye using `ofAction` operartors. This is not very straighforward and requires multiple LOCs to get it setup.

### Describe the solution you'd like
A possible solution is to be able to filter the lifecycle in the `@Action` decorator, so you can hook onto the specific event you need

```
@Action(ofActionSuccesful(MyAction))
myAction(ctx: StateContext){

}

@Action(ofActionErrored(MyAction))
myAction(ctx: StateContext){

}

...
```

### Describe alternatives you've considered
There's been some discussion on [slack](https://ngxs.slack.com/archives/CA0NS79MG/p1598955107012900), and here's a [gist](https://gist.github.com/enko/3514a3223b1374f7ad13bd479579148c) @timdeschryver proposed

**Comment by timdeschryver** [comment]
👋 I think you pinged the wrong Tim 😅

**Comment by enko** [comment]
That would be me 😉

**Comment by enko** [comment]
Some implementation notes from my side:

The decorator `@Action` is already in use, so I choose `@Actions` for my quick hack. I think `@ActionHook` or something like that, would be more appropriate. You need to wrap the `ofActionSuccessful` inside a function, because at the time the decorators are resolved the internal state of ngxs is not completely build up, so the function call `ofActionSuccessfull(FooAction)` will fail.

@markwhitfeld would you be open for a pull request or would you see this more in an extra module?
